doc/source/rllib/rllib-offline.rst

```diff
@@ -8,217 +8,662 @@ Working With Offline Data
+During online training, RLlib uses `SingleAgentEnvRunner <rllib-concepts.html#policy-evaluation>`__ actors to generate episodes 
+of experiences in parallel using the current policy. RLlib can also uses this same episode format for reading and writing 
+experiences to offline storage. For offline writing RLlib uses the `OfflineSingleAgentEnvRunner 
+<https://github.com/ray-project/ray/blob/master/rllib/offline/offline_single_agent_env_runner.py>`__ that samples episodes 
+and writes them to offline storage. 
+
+You can store experiences either directly in RLlib's episode format or in table (columns) 
+format. You should use the episode format when 
+
+a. You need experiences grouped by their trajectory and ordered in time (e.g. to train stateful modules).
+b. You want to use recorded experiences exclusively within RLlib.
+
+Contrary, you should prefer the table (columns) format, if
+
+a. You need to read the data easily with other data tools or ML libraries.
+b. You do not need full (usually complete) trajectories in training.
+
+.. note:: RLlib's redesigned API stack incorporates principles that support standalone applications. Consequently, the 
+    `SingleAgentEpisode` class is available for use independently of RLlib. To enable faster access through external data tools 
+    (e.g., for data transformations), it is recommended to use the table record format in these scenarios.
+
+Most importantly, RLlib's offline RL API builds on top of :ref:`Ray Data <data>` and therefore features in general all read and 
+write methods supported by Ray Data (e.g. ``read_parquet``, ``read_json``, etc.) with ``read/write_parquet`` being the default 
+read and write methods. A core design principle of the API is to apply as many data transformations as possible on-the-fly to
+the learner, allowing the learner to focus exclusively on model updates. 
+
+.. note:: During the transition phase from old to new API stack you can use the new offline RL API also with your 
+    `SampleBatch <https://github.com/ray-project/ray/blob/master/rllib/offline/offline_single_agent_env_runner.py>`__ data recorded 
+    with the old API stack. To enable this feature set ``input_read_sample_batches=True``.
+
+Example: Training an expert policy.
+--------------------------------------------
+In this example you train an PPO agent to play ``CartPole-v1`` until it collects an episode mean reward of ``450.0``. You checkpoint 
+this agent and later use its policy to record expert data to local disk.
+
+.. testsetup:: 
+    
+    # Define a shared variable to store the path to the
+    # best checkpoint.
+    best_checkpoint = None
+
+    # Define a shared variable to store the path to the
+    # recorded data.
+    data_path = None
+
+    # Define another shared variable to store the path to
+    # the tabular recording data.
+    tabular_data_path = None
+
+.. testcode::
```

Comment:
remove this line?

Reply:
Nope, the `testsetup` is hidden for users. We define there only some global variables we can use across all examples :)